iothub_client/samples/pnp/pnp_temperature_controller/pnp_thermostat_component.c

```diff
@@ -0,0 +1,367 @@
+    else if ((responseBuilder = calloc(1, responseBuilderSize + 1)) == NULL)
+    {
+        LogError("Unable to allocate %lu bytes", (unsigned long)(responseBuilderSize + 1));
+        result = false;
+    }
+    else if ((responseBuilderSize = snprintf((char*)responseBuilder, responseBuilderSize + 1, g_minMaxCommandResponseFormat, pnpThermostatComponent->minTemperature, pnpThermostatComponent->maxTemperature, 
+                                              pnpThermostatComponent->allTemperatures / pnpThermostatComponent->numTemperatureUpdates, g_programStartTime, currentTime)) < 0)
+    {
+        LogError("snprintf to output buffer for command response");
+        result = false;
+    }
+    else
+    {
+        result = true;
+    }
+
+    if (result == true)
+    {
+        *response = responseBuilder;
+        *responseSize = (size_t)responseBuilderSize;
+    }
+    else
+    {
+        free(responseBuilder);
+    }
+
+    return result;
+}       
+
+int PnP_ThermostatComponent_ProcessCommand(PNP_THERMOSTAT_COMPONENT_HANDLE pnpThermostatComponentHandle, const char *pnpCommandName, JSON_Value* commandJsonValue, unsigned char** response, size_t* responseSize)
+{
+    PNP_THERMOSTAT_COMPONENT* pnpThermostatComponent = (PNP_THERMOSTAT_COMPONENT*)pnpThermostatComponentHandle;
+    char* jsonStr = NULL;
+    const char* sinceStr;
+    int result;
+
+    *response = NULL;
+    *responseSize = 0;
+
+    if (strcmp(pnpCommandName, g_GetMinMaxReport) != 0)
+    {
+        LogError("PnP command=%s is not supported on thermostat component", pnpCommandName);
+        result = PNP_STATUS_NOT_FOUND;
+    }
+    // See caveats section in ../readme.md; we don't actually respect this sinceStr to keep the sample simple,
+    // but want to demonstrate how to parse out in any case.
+    else if ((sinceStr = json_value_get_string(commandJsonValue)) == NULL)
+    {
+        LogError("Cannot retrieve JSON string for command");
+        result = PNP_STATUS_BAD_FORMAT;
+    }
+    else if (BuildMaxMinCommandResponse(pnpThermostatComponent, response, responseSize) == false)
+    {
+        LogError("Unable to build response for component=%s", pnpThermostatComponent->componentName);
+        result = PNP_STATUS_INTERNAL_ERROR;
+    }
+    else
+    {
+        LogInfo("Returning success from command request for component=%s", pnpThermostatComponent->componentName);
+        result = PNP_STATUS_SUCCESS;
+    }
+
+    free(jsonStr);
+    return result;
+}
+
+//
+// UpdateTemperatureAndStatistics updates the temperature and min/max/average values
+//
+static void UpdateTemperatureAndStatistics(PNP_THERMOSTAT_COMPONENT* pnpThermostatComponent, double desiredTemp, bool* maxTempUpdated)
+{
+    if (desiredTemp > pnpThermostatComponent->maxTemperature)
+    {
+        pnpThermostatComponent->maxTemperature = desiredTemp;
+        *maxTempUpdated = true;
```

Comment:
>maxTempUpdated [](start = 9, length = 14)

lets make maxTempUpdated = false in all other cases

Reply:
yes i like this layer not putting assumptions on caller's state.  Fixing.